kiwi_lint/__init__.py

```diff
@@ -34,3 +35,12 @@ def register(linter):
     linter.register_checker(NestedDefinitionChecker(linter))
     linter.register_checker(MissingPermissionsChecker(linter))
     linter.register_checker(AutoFieldChecker(linter))
+    linter.register_checker(
+        FunctionBasedViewChecker(
+            linter,
+            main_urls_package='tcms.urls',
+            filters=[
+                lambda apps: filter(lambda app: app.startswith('tcms'), apps)
```

Comment:
Why exactly do we need to have filters ? I tried without it and everything seems to work just fine.

From what I can tell the filters will only limit the list of modules which the checker collects but that has nothing to do with the list of python files which pylint inspects. That is if I don't specify filters I get several extra view modules in the data structure (e.g. modernrpc.views) but because pylint is inspecting only files in the tcms/ directory I never get to do a lookup for these keys in the dictionary.

IMO you can remove everything related to filters and simplify even more.

Reply:
Good call, I did not take into account that you invoke pylint on specific files.